command/eval_list.go

```diff
@@ -0,0 +1,182 @@
+package command
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/nomad/api"
+	"github.com/hashicorp/nomad/api/contexts"
+	"github.com/posener/complete"
+)
+
+type EvalListCommand struct {
+	Meta
+}
+
+func (c *EvalListCommand) Help() string {
+	helpText := `
+Usage: nomad eval list [options]
+
+  List is used to list the set of evaluations processed by Nomad.
+
+General Options:
+
+  ` + generalOptionsUsage(usageOptsDefault) + `
+
+Eval Status Options:
+
+  -verbose
+    Show full information.
+
+  -per_page
+    How many results to show per page.
+
+  -page_token
+    Where to start pagination.
+
+  -job
+    Only show evaluations for this job ID.
+
+  -status
+    Only show evaluations with this status.
+
+  -json
+    Output the evaluation in its JSON format.
+
+  -t
+    Format and display evaluation using a Go template.
+`
+
+	return strings.TrimSpace(helpText)
+}
+
+func (c *EvalListCommand) Synopsis() string {
+	return "Display evaluation status and placement failure reasons"
+}
+
+func (c *EvalListCommand) AutocompleteFlags() complete.Flags {
+	return mergeAutocompleteFlags(c.Meta.AutocompleteFlags(FlagSetClient),
+		complete.Flags{
+			"-json":       complete.PredictNothing,
+			"-t":          complete.PredictAnything,
+			"-verbose":    complete.PredictNothing,
+			"-job":        complete.PredictAnything,
+			"-status":     complete.PredictAnything,
+			"-per_page":   complete.PredictAnything,
+			"-page_token": complete.PredictAnything,
+		})
+}
+
+func (c *EvalListCommand) AutocompleteArgs() complete.Predictor {
+	return complete.PredictFunc(func(a complete.Args) []string {
+		client, err := c.Meta.Client()
+		if err != nil {
+			return nil
+		}
+
+		if err != nil {
+			return nil
+		}
+
+		resp, _, err := client.Search().PrefixSearch(a.Last, contexts.Evals, nil)
+		if err != nil {
+			return []string{}
+		}
+		return resp.Matches[contexts.Evals]
+	})
+}
+
+func (c *EvalListCommand) Name() string { return "eval status" }
+
+func (c *EvalListCommand) Run(args []string) int {
+	var monitor, verbose, json bool
+	var perPage int
+	var tmpl, pageToken, filterJobID, filterStatus string
+
+	flags := c.Meta.FlagSet(c.Name(), FlagSetClient)
+	flags.Usage = func() { c.Ui.Output(c.Help()) }
+	flags.BoolVar(&monitor, "monitor", false, "")
+	flags.BoolVar(&verbose, "verbose", false, "")
+	flags.BoolVar(&json, "json", false, "")
+	flags.StringVar(&tmpl, "t", "", "")
+	flags.IntVar(&perPage, "per_page", 0, "")
+	flags.StringVar(&pageToken, "page_token", "", "")
+	flags.StringVar(&filterJobID, "job", "", "")
+	flags.StringVar(&filterStatus, "status", "", "")
+
+	if err := flags.Parse(args); err != nil {
+		return 1
+	}
+
```

Comment:
I can't think of any other command where we do this, but should we check that no extra args were passed and show the help text? Currently it will just run:

```console
$ nomad eval list abc
ID        Priority  Triggered By        Job ID                                          Status    Placement Failures
00357336  50        node-update         sysbatch-for-real/dispatch-1639519858-f98cbccb  complete  false
9efcd819  50        node-update         example                                         complete  false
f4518197  50        node-update         example                                         complete  false
f719bcc0  50        node-update         sysbatch-for-real/dispatch-1639519858-f98cbccb  complete  false
319bbc6d  50        job-register        sysbatch-for-real/dispatch-1639519858-f98cbccb  complete  false
b1f4ed93  50        deployment-watcher  example                                         complete  false
326f13dd  50        job-register        example                                         complete  false
```

Reply:
Yes, we check it in `nomad deployment list`! So let's do that here too.